front/app/utils/participationMethodUtils.tsx

```diff
@@ -1,25 +1,76 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
+
+// intl
+import { FormattedMessage } from './cl-intl';
+import messages from './participationMethodUtilsMessages';
+
+// services
 import {
-  getInputTerm,
   ParticipationMethod,
+  getInputTerm,
 } from 'services/participationContexts';
 import { IPhaseData } from 'services/phases';
 import { IProjectData } from 'services/projects';
-import { FormattedMessage } from 'utils/cl-intl';
-import messages from './participationMethodUtilsMessages';
+
+// components
+import SharingModalContent from 'components/PostShowComponents/SharingModalContent';
+
+// utils
 import { isNilOrError } from './helperUtils';
+import clHistory from 'utils/cl-router/history';
+import { IIdea } from 'services/ideas';
 
 type ParticipationMethodConfig = {
   /** We currently have 2 UIs for admins to edit the form definition. This
    * defines which UI, if any, the method uses */
   formEditor: 'simpleFormEditor' | 'surveyEditor' | null;
-  getFormTitle?: any;
+  onFormSubmission: (
+    _project?: IProjectData,
+    ideaId?: string,
+    idea?: IIdea,
+    phaseId?: string
+  ) => void;
+  getModalContent: () => ReactNode | null;
+  getFormTitle?: (_project: IProjectData, _phases: [IPhaseData]) => void;
   showInputManager: boolean;
 };
 
 const ideationConfig: ParticipationMethodConfig = {
   formEditor: 'simpleFormEditor',
-  getFormTitle: (project, phases) => {
+  onFormSubmission: (
```

Comment:
Minor: Since the types are similar and reused, we could use an object type and reuse that where needed such that it is easier to track when changes to the type are made.